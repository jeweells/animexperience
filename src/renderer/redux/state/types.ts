import { FStatus, Optional, TopView } from '@shared/types'
import {
  AnimeInfo,
  DeepAnimeIdSearchResult,
  EpisodeInfo,
  FollowedAnime,
  RecentAnimeInfo,
  RecommendationInfo,
  WatchHistoryItem
} from '@shared/types'
import { FollowedAnimeWStatus } from '../reducers/followedAnimes'
import { PreAllowWatch } from '../reducers/invokedLink'
import { RecentAnimeData } from '../../src/hooks/useRecentAnimes'
import { VideoOption } from '../../src/components/VideoPlayer'

export type AnimeSearchState = {
  // Boolean id
  searching?: string
  result?: Optional<DeepAnimeIdSearchResult>
  status: Partial<{
    result: FStatus
    moreResults: FStatus
  }>
}

export type FollowedAnimesState = {
  followedDict: Record<string, FollowedAnime>
  followed: FollowedAnimeWStatus[]
  status: Partial<{
    followed: FStatus
  }>
}

export type InvokedLinkState = {
  open: {
    watch?: string
  }
  preAllow: Partial<{
    watch: PreAllowWatch
  }>
  status: {
    watch?: FStatus
  }
}

export type NotificationsState = {
  message: string | null
  key: string
}

export type PeekState = {
  name?: string
  status: Partial<{
    info: FStatus
  }>
  info?: Optional<AnimeInfo>
  peeking?: string
}

export type PlayerState = {
  open?: string
  freezed?: boolean
}

export type OptionInfo = {
  name: string
  prefer?: boolean
  score: number
}
// Define a type for the slice state
export type PlayerOptionsState = {
  options?: OptionInfo[]
  history?: string[]
  preferred?: string[]
  status: Partial<{
    options: FStatus
  }>
}

export type RecommendationsState = {
  [name: string]: Partial<{
    status: FStatus
    recommendations: RecommendationInfo[]
  }>
}

export type TopviewState = {
  views: TopView[]
}

export type WatchState = {
  watching?: Optional<RecentAnimeData>
  availableVideos?: Optional<VideoOption[]>
  status: {
    availableVideos?: FStatus
    info?: FStatus
    watchEpisode?: FStatus
  }
  info?: Optional<AnimeInfo>
  showNextEpisodeButton?: boolean
  nextEpisodeTimeout: number
  autoFullScreen?: boolean
}

export type WatchedState = {
  episodes: {
    [name: string]: EpisodeInfo
  }
  recently?: RecentAnimeInfo[]
  status: {
    episodes?: FStatus
    recently?: FStatus
  }
}

export type WatchHistoryState = {
  sorted?: WatchHistoryItem[]
  status: {
    sorted?: FStatus
  }
}
