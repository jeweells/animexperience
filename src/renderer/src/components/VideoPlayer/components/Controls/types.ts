export type ControlsContext = {
  video: HTMLVideoElement | null
}
