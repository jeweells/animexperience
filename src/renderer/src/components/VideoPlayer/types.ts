import { RecentAnimeData } from '../../hooks/useRecentAnimes'
import { Optional } from '@shared/types'
import { VideoOption } from './index'

export type $IframeContents = JQuery<HTMLIFrameElement | Text | Comment | Document>
export type BasicVideoInfo = {
  option?: Optional<VideoOption>
  anime?: Optional<RecentAnimeData>
}

export type KnownOption =
  | 'mega'
  | 'fembed'
  | 'okru'
  | 'mixdrop'
  | 'streamtape'
  | 'stape'
  | 'maru'
  | 'sw'
  | 'yourupload'
