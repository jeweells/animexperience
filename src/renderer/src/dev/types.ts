import { RawDevMessage } from '@shared/types'
import { Moment } from 'moment'

export type DevMessage = RawDevMessage & {
  id: string
  at: Moment
}
