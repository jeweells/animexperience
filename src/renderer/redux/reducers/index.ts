export { animeSearch } from './animeSearch'
export { followedAnimes } from './followedAnimes'
export { invokedLink } from './invokedLink'
export { notifications } from './notifications'
export { peek } from './peek'
export { player } from './player'
export { playerOptions } from './playerOptions'
export { recommendations } from './recommendations'
export { topView } from './topView'
export { watch } from './watch'
export { watched } from './watched'
export { watchHistory } from './watchHistory'
