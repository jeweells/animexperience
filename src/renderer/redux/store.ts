import { configureStore } from '@reduxjs/toolkit'
import animeSearch from './reducers/animeSearch'
import followedAnimes from './reducers/followedAnimes'
import peek from './reducers/peek'
import player from './reducers/player'
import playerOptions from './reducers/playerOptions'
import recommendations from './reducers/recommendations'
import topView from './reducers/topView'
import watch from './reducers/watch'
import watched from './reducers/watched'
import watchHistory from './reducers/watchHistory'
import invokedLink from './reducers/invokedLink'
import notifications from './reducers/notifications'

export default configureStore({
  reducer: {
    watch,
    player,
    watched,
    recommendations,
    watchHistory,
    peek,
    topView,
    playerOptions,
    animeSearch,
    followedAnimes,
    invokedLink,
    notifications
  }
})
