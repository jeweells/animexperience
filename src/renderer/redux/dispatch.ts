import { AppDispatch } from '~/redux/utils'
import store from '~/redux/store'

export const dispatch: AppDispatch = (action) => store.dispatch(action)
