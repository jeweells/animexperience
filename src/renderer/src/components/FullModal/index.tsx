import Dialog, { DialogProps } from '@mui/material/Dialog'
import Fade from '@mui/material/Fade'
import { useLayoutEffect, FC, memo } from 'react'
import { styled } from '@mui/system'
import { topView } from '@reducers'
import { useAppDispatch, useAppSelector } from '~/redux/utils'
import { TopView } from '@shared/types'
import { useTopBarHeight } from '../Topbar'
import { useKeyUp } from '~/src/hooks/useKeyboardKeys'

interface SModalProps extends DialogProps {
  topBarHeight: number
  contrast?: boolean
}

const SModal = styled(Dialog, {
  shouldForwardProp(propName) {
    return !['contrast', 'topBarHeight'].includes(propName as string)
  }
})<SModalProps>`
  margin: 0;
  width: 100vw;
  overflow: hidden;
  --modal-height: calc(100vh - ${(props) => props.topBarHeight}px);
  height: var(--modal-height);
  will-change: height, top;
  background-color: transparent;
  .MuiDialog-container {
    background-color: #0f131a;
  }
  top: ${(props) => props.topBarHeight}px !important;
  .MuiPaper-root {
    background-image: unset;
    background-color: ${(props) => (props.contrast ? '#0f131a' : 'unset')};
  }
`

export interface FullModalProps extends Omit<DialogProps, 'open' | 'sx'> {
  contrast?: boolean
  show: string | undefined
  view: TopView
  onPopRequested?: () => void
}

export const FullModal: FC<FullModalProps> = memo<FullModalProps>(
  ({ view, children, show, onPopRequested, ...rest }) => {
    const topBarHeight = useTopBarHeight()
    const currentTopview = useAppSelector((d) => d.topView.views[0])
    const dispatch = useAppDispatch()

    useLayoutEffect(() => {
      if (show) {
        dispatch(topView.push(view))
      } else {
        dispatch(topView.pop(view))
      }
    }, [show])

    useKeyUp(
      () => {
        if (currentTopview !== view) return
        onPopRequested?.()
      },
      { code: 'Escape' }
    )

    return (
      <SModal
        aria-labelledby={'full-modal'}
        TransitionComponent={Fade}
        fullScreen={true}
        disableEnforceFocus={true}
        topBarHeight={topBarHeight}
        open={!!currentTopview && currentTopview === view}
        {...rest}
        TransitionProps={{
          appear: true,
          mountOnEnter: true,
          unmountOnExit: true,
          timeout: 300,
          ...rest.TransitionProps,
          onExited(arg: HTMLElement) {
            // When show is valid the view is pushed into the topViews and the modal should not be
            // considered as closed
            // TODO: avoid calling the callbacks: onClose, onExiting
            if (!show) {
              rest.TransitionProps?.onExited?.(arg)
            }
          }
        }}
      >
        {children}
      </SModal>
    )
  }
)

FullModal.displayName = 'FullModal'

export default FullModal
