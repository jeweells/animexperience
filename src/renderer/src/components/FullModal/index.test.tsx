import FullModal, { FullModalProps } from './index'
import { Provider } from 'react-redux'
import { RootState } from '~/redux/state'
import { DeepPartial } from 'redux'
import { TopView } from '@shared/types'
import { MockStoreEnhanced } from 'redux-mock-store'
import theme from '../../theme'
import { ThemeProvider } from '@mui/material'

describe('FullModal', () => {
  let initialState: DeepPartial<RootState>
  let props: FullModalProps
  let store: MockStoreEnhanced

  beforeEach(() => {
    props = {
      show: 'sample-id',
      view: TopView.SEARCH,
      contrast: false
    }
    initialState = {
      topView: {
        views: [TopView.SEARCH]
      }
    }
  })

  const getComponent = (override: Partial<FullModalProps> = {}) => {
    store = mockStore(initialState)
    return (
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <FullModal {...props} {...override} />
        </ThemeProvider>
      </Provider>
    )
  }

  it('renders default', async () => {
    const wrapper = render(getComponent())
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders default with contrast', async () => {
    const wrapper = render(getComponent({ contrast: true }))
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders when another view is active', async () => {
    initialState.topView!.views = [TopView.PEEK]
    const wrapper = render(getComponent())
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders when not showing', async () => {
    const wrapper = render(getComponent({ show: undefined }))
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('showing pushes view when mounted', async () => {
    render(getComponent())
    expect(store.getActions()).toContainEqual(
      expect.objectContaining({
        payload: props.view,
        type: 'topView/push'
      })
    )
  })

  it('not showing pops view when mounted', async () => {
    render(getComponent({ show: undefined }))
    expect(store.getActions()).toContainEqual(
      expect.objectContaining({
        payload: props.view,
        type: 'topView/pop'
      })
    )
  })
})
