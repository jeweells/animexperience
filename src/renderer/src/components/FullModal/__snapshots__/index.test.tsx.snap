// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullModal renders default 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  margin: 0;
  width: 100vw;
  overflow: hidden;
  --modal-height: calc(100vh - 56px);
  height: var(--modal-height);
  will-change: height,top;
  background-color: transparent;
  top: 56px!important;
}

@media print {
  .emotion-0 {
    position: absolute!important;
  }
}

.emotion-0 .MuiDialog-container {
  background-color: #0f131a;
}

.emotion-0 .MuiPaper-root {
  background-image: unset;
  background-color: unset;
}

.emotion-1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-2 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-2 {
    height: auto;
  }
}

.emotion-3 {
  background-color: #121212;
  color: #e9ebf0;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);
  background-image: linear-gradient(rgba(255, 255, 255, 0.16), rgba(255, 255, 255, 0.16));
  margin: 0;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: none;
  max-width: 100%;
  width: 100%;
  height: 100%;
  border-radius: 0;
}

@media print {
  .emotion-3 {
    overflow-y: visible;
    box-shadow: none;
  }
}

.emotion-3.MuiDialog-paperScrollBody {
  margin: 0;
  max-width: 100%;
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop emotion-1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper emotion-2"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="full-modal"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen emotion-3"
        role="dialog"
      />
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`FullModal renders default with contrast 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  margin: 0;
  width: 100vw;
  overflow: hidden;
  --modal-height: calc(100vh - 56px);
  height: var(--modal-height);
  will-change: height,top;
  background-color: transparent;
  top: 56px!important;
}

@media print {
  .emotion-0 {
    position: absolute!important;
  }
}

.emotion-0 .MuiDialog-container {
  background-color: #0f131a;
}

.emotion-0 .MuiPaper-root {
  background-image: unset;
  background-color: #0f131a;
}

.emotion-1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-2 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-2 {
    height: auto;
  }
}

.emotion-3 {
  background-color: #121212;
  color: #e9ebf0;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);
  background-image: linear-gradient(rgba(255, 255, 255, 0.16), rgba(255, 255, 255, 0.16));
  margin: 0;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: none;
  max-width: 100%;
  width: 100%;
  height: 100%;
  border-radius: 0;
}

@media print {
  .emotion-3 {
    overflow-y: visible;
    box-shadow: none;
  }
}

.emotion-3.MuiDialog-paperScrollBody {
  margin: 0;
  max-width: 100%;
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop emotion-1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper emotion-2"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="full-modal"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen emotion-3"
        role="dialog"
      />
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`FullModal renders when another view is active 1`] = `
<body
  style=""
>
  <div />
</body>
`;

exports[`FullModal renders when not showing 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  margin: 0;
  width: 100vw;
  overflow: hidden;
  --modal-height: calc(100vh - 56px);
  height: var(--modal-height);
  will-change: height,top;
  background-color: transparent;
  top: 56px!important;
}

@media print {
  .emotion-0 {
    position: absolute!important;
  }
}

.emotion-0 .MuiDialog-container {
  background-color: #0f131a;
}

.emotion-0 .MuiPaper-root {
  background-image: unset;
  background-color: unset;
}

.emotion-1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-2 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-2 {
    height: auto;
  }
}

.emotion-3 {
  background-color: #121212;
  color: #e9ebf0;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);
  background-image: linear-gradient(rgba(255, 255, 255, 0.16), rgba(255, 255, 255, 0.16));
  margin: 0;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: none;
  max-width: 100%;
  width: 100%;
  height: 100%;
  border-radius: 0;
}

@media print {
  .emotion-3 {
    overflow-y: visible;
    box-shadow: none;
  }
}

.emotion-3.MuiDialog-paperScrollBody {
  margin: 0;
  max-width: 100%;
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop emotion-1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper emotion-2"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="full-modal"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen emotion-3"
        role="dialog"
      />
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
